taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -516,17 +517,21 @@ func (self *Contract) DistributeRewards(blockAuthorAddr *common.Address, rewards
 
 	votesReward := big.NewInt(0)
 	blockAuthorReward := big.NewInt(0)
-	trxsReward := blockReward
+	DAGProposersReward := blockReward
 	// We need to handle case for block 1
 	if rewardsStats.TotalVotesWeight > 0 {
 		// Calculate propotion between votes and transactions
-		trxsReward = bigutil.Div(bigutil.Mul(blockReward, TransactionsRewardPercentage), big.NewInt(100))
-		votesReward = bigutil.Sub(blockReward, trxsReward)
+		DAGProposersReward = bigutil.Div(bigutil.Mul(blockReward, DAGRewardPercentage), big.NewInt(100))
+		votesReward = bigutil.Sub(blockReward, DAGProposersReward)
 
 		// Calculate bonus reward based on MaxBlockAuthorReward and subtract from total votes reward
 		bonusReward := bigutil.Div(bigutil.Mul(votesReward, big.NewInt(int64(self.cfg.MaxBlockAuthorReward))), big.NewInt(100))
 		votesReward = bigutil.Sub(votesReward, bonusReward)
 
+		if bigutil.Add(bigutil.Add(DAGProposersReward, bonusReward), votesReward).Cmp(blockReward) != 0 {
+			panic("DistributeRewards - wrong reward parts calculations")
```

Comment:
how is this possible ? When I check the math if the percentage from config are set in their boundaries, e.g. not > 100% etc... this can never happen